components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -2343,6 +2327,64 @@ public static String createEndpoint(String endpointUrl, Registry registry) throw
         return externalAPIStores;
     }
 
+    /**
+     * Get OMElement iterator for external stores configured in external-store.xml in tenant registry.
+     *
+     * @param tenantId Tenant ID
+     * @return ExternalStores OMElement Iterator
+     * @throws APIManagementException If an error occurs while reading external-store.xml
+     */
+    private static Iterator getExternalStoresIteratorFromConfig(int tenantId) throws APIManagementException {
+        Iterator apiStoreIterator = null;
+        try {
+            UserRegistry registry = ServiceReferenceHolder.getInstance().getRegistryService()
+                    .getGovernanceSystemRegistry(tenantId);
+            if (registry.resourceExists(APIConstants.EXTERNAL_API_STORES_LOCATION)) {
+                Resource resource = registry.get(APIConstants.EXTERNAL_API_STORES_LOCATION);
+                String content = new String((byte[]) resource.getContent(), Charset.defaultCharset());
+                OMElement element = AXIOMUtil.stringToOM(content);
+                apiStoreIterator = element.getChildrenWithLocalName("ExternalAPIStore");
+            }
+        } catch (RegistryException e) {
+            String msg = "Error while retrieving External Stores Configuration from registry";
+            log.error(msg, e);
```

Comment:
do we need to throw and log same exception?

Reply:
The thrown exception is not logged in server from any of the calling methods, hence we need to log it right?

Reply:
Found that the error is logged in GlobalThrowableMapper from [1] and [2].

[1] https://github.com/wso2/carbon-apimgt/blob/5fef47dff0a057eaaa88993ada0497db3db7286d/components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/exception/GlobalThrowableMapper.java#L168

[2]https://github.com/wso2/carbon-apimgt/blob/5fef47dff0a057eaaa88993ada0497db3db7286d/components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/exception/GlobalThrowableMapper.java#L179

Hence removed the additional error logs from APIUtil.